src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -67,6 +78,20 @@ public void resetData(ReadOnlyAddressBook newData) {
         indicateAddressBookChanged();
     }
 
+    @Override
+    public void resetData(ReadOnlyClubBudgetElementsBook newData) {
+        searchHistoryManager.clearSearchHistory();
+        versionedClubBudgetElementsBook.resetData(newData);
+        indicateClubBudgetElementsBookChanged();
+    }
+
+    @Override
+    public void resetData(ReadOnlyFinalBudgetBook newData) {
+        searchHistoryManager.clearSearchHistory();
```

Comment:
I don't think there is a need to clear search history over here since searchHistoryManager is only relevant to the Persons List. Did i miss something out?